### DISCUSSION TITLE
TTGO T-Display-S3 and missing i80 lcd interface support

### DISCUSSION THREAD

**Discussion by wizmo2** [sle118/squeezelite-esp32#296] [Ideas]
I have a TTGO T-Display-S3 and have been able to flash it.   Its a nice platform with a 1.9in ST7789 display, but its an i80 interface.  

Was looking at what was involved in adding support to the existing GDS system.  Think it's probably do-able, just a matter of how far down the rabbit hole it ends up going.

**Comment by sle118** [comment]
I'll speak for myself, but I think we're at a point where we might consider keeping support (bug fixes)  for the legacy platform, which will continue to attract users thanks to its Bluetooth audio, but take the opportunity to position ourselves in such way that we can truely leverage the power of the S3 platform, which is a real evolution over the venerable WROVER. Amongst others, the minimum flash size should be carefully be evaluated as I really like to build a real OTA mechanism instead of the current recovery/application partition design which was dictated by the common 4mb flash a couple of years ago.

**Reply by wizmo2** [reply]
For me the lack of Bluetooth source functionality (for portable devices) is a a potential issues, but my assumption is that Espressif will address this with some form of BLE 5 audio support in the future.
My drive has always been LMS, so BT Sink/Airplay/Cspot is not as important for me, especially now you can get budget d-class amps with BT receiver built in.

**Reply by philippe44** [reply]
As long as I80 supports the idea of WriteCommand and WriteData, then it should not bee too painful. You need to add an display/core/ifaces with a few methods, handle it's initialization in display/display.c and all the rest should follow.

[edit]: it seems that I80 has the same D/C concept than SPI do it's probably fairly doable.
